### ISSUE TITLE
Perform GC in parallel

### ISSUE THREAD

**Issue by vito** [concourse/concourse#4152]
In #3602 we noticed that during a `VACUUM` container counts would climb up and max out the workers.

This seems to imply that GC was held up somewhere. If containers are being created, queries are running to create those, so it's not like the whole table is locked up.

We currently run all of our collectors one after another: https://github.com/concourse/concourse/blob/8c5d0dc1a9c658b7ac8e978716d9d4d008707281/atc/gc/collector.go#L51

Maybe one of the collectors was blocked waiting on the `VACUUM`? Should we run these in parallel on independent loops instead?

Can we confirm the slowdown by observing logs on the `algorithm` environment? (This is time-sensitive, since they don't currently aggregate anywhere.)

**Comment by vito** [comment]
Updating the title. The investigation led us to parallelizing GC on the algorithm branch, which proved pretty effective at fixing this. The main problem was that some collectors would just take a long time, and while they were GC'ing nothing else could GC.

There was a solid theory to the order that we would do GC sequentially, but doing them in parallel should be fine too - it just may take multiple ticks for objects to expire at times. The improved resiliency seems worth the trade-off.

Side note: we did notice that, on occasion, a single worker would still have its container count spike up. This seems to be a separate issue. With our fix implemented we never saw *all* workers spike up again, so it's still a valid fix. We dug into it and didn't really find a root cause. Container destroying was just taking longer - up to a couple minutes in some cases, whereas typically it takes a second or two. We don't have an explanation for the slowdown, but it was obvious that if container destroying was taking that long that it would snowball - there were 131 containers to GC in one tick, and if each one takes a couple minutes, it's not fast enough to overcome the rate of container creation in our test environment.

**Comment by vito** [comment]
Also: would longer/shorter intervals make sense for some garbage-collectors vs. others?
